src/test/java/hudson/tasks/junit/pipeline/JUnitResultsStepTest.java

```diff
@@ -245,26 +246,36 @@ public void testTrends() throws Exception {
         WorkflowJob j = rule.jenkins.createProject(WorkflowJob.class, "testTrends");
         j.setDefinition(new CpsFlowDefinition("node {\n" +
                 "  stage('first') {\n" +
-                "    def first = junit(testResults: \"junit-report-testTrends-first-${env.BUILD_NUMBER}.xml\")\n" +
+                "    def first = junit(testResults: \"junit-report-testTrends-first.xml\")\n" +
                 "  }\n" +
                 "  stage('second') {\n" +
-                "    def second = junit(testResults: \"junit-report-testTrends-second-${env.BUILD_NUMBER}.xml\")\n" +
+                "    def second = junit(testResults: \"junit-report-testTrends-second.xml\")\n" +
                 "  }\n" +
                 "}\n", true));
         FilePath ws = rule.jenkins.getWorkspaceFor(j);
-        for (int i = 1; i < 4; i++) {
-            FilePath firstFile = ws.child("junit-report-testTrends-first-" + i + ".xml");
-            firstFile.copyFrom(JUnitResultsStepTest.class.getResource("junit-report-testTrends-first-" + i + ".xml"));
-            FilePath secondFile = ws.child("junit-report-testTrends-second-" + i + ".xml");
-            secondFile.copyFrom(JUnitResultsStepTest.class.getResource("junit-report-testTrends-second-" + i + ".xml"));
-        }
+        FilePath firstFile = ws.child("junit-report-testTrends-first.xml");
+        FilePath secondFile = ws.child("junit-report-testTrends-second.xml");
+
+        // Populate first run's tests.
+        firstFile.copyFrom(JUnitResultsStepTest.class.getResource("junit-report-testTrends-first-1.xml"));
+        secondFile.copyFrom(JUnitResultsStepTest.class.getResource("junit-report-testTrends-second-1.xml"));
```

Comment:
Maybe it would be handy for there to be a `Step` defined in some `*-tests.jar` that copied a classpath resource to the workspace, so that tests like this would not need to call `getWorkspaceFor` and could use something like

```groovy
stage('first') {
  writeTestResourceToWorkspace resource: '/some/pkg/junit-report-testTrends-first-1.xml', file: 'junit-report-testTrends-first.xml'
  junit 'junit-report-testTrends-first.xml'
}